account-compression/programs/account-compression/src/macros.rs

```diff
@@ -0,0 +1,106 @@
+/// This macro applies functions on a ConcurrentMerkleT:ee and emits leaf information
+/// needed to sync the merkle tree state with off-chain indexers.
+#[macro_export]
+macro_rules! merkle_tree_depth_size_apply_fn {
+    ($max_depth:literal, $max_size:literal, $id:ident, $bytes:ident, $func:ident, true, $($arg:tt)*)
```

Comment:
nit take-it-or-leave-it: rather than specifying `true` and `false` for mutability and matching on that, how about a private enum like
```
enum TreeLoad {
    Immutable,
    Mutable,
}
```

Reply:
AFAIK macros can't do runtime expansions like matching over an enum. True/false for mutability is really just SDK ergonomics.

Reply:
You're doing a compile-time match, not a runtime match. So you can do:

```
enum TreeLoad {
      Immutable,
      Mutable,
  }

  macro_rules! apply_fn {
      ($func:ident, TreeLoad::Immutable, $($arg:tt)*) => {
          println!("immutable");
          $func($($arg)*)
      };
      ($func:ident, TreeLoad::Mutable, $($arg:tt)*) => {
          println!("mutable");
          $func($($arg)*)
      };
  }

  fn do_print(word: &str) {
      println!("Hello, {word}!");
  }

  fn main() {
      apply_fn!(do_print, TreeLoad::Immutable, "immutable hi");
      apply_fn!(do_print, TreeLoad::Mutable, "mutable hi");
  }
```

Although weirdly enough, it says that `TreeLoad` is never used, so :shrug:

Reply:
Awesome, thanks! Updated with the Enum